collections/nemo_nlp/nemo_nlp/modules/losses.py

```diff
@@ -184,19 +185,37 @@ def create_ports():
         }
         return input_ports, output_ports
 
-    def __init__(self, num_slots, **kwargs):
+    def __init__(self,
+                 num_slots,
+                 slot_classes_loss_weights=None,
+                 intent_classes_loss_weights=None,
+                 intent_loss_weight=0.6,
+                 **kwargs):
         LossNM.__init__(self, **kwargs)
         self.num_slots = num_slots
-        self._criterion = nn.CrossEntropyLoss()
+        self.intent_loss_weight = intent_loss_weight
+
+        # For weighted loss to tackle class imbalance
+        if len(slot_classes_loss_weights) > 0:
+            self.slot_classes_loss_weights = \
+                torch.FloatTensor(slot_classes_loss_weights).cuda()
```

Comment:
1. do we have to make sure that weights have the same number of elements as the number of slots?
2. don't just move it to cuda bc it won't be compatible on cpus. use self._device and move it to self._device

Reply:
1. This is done automatically with the online calculation of weights unless the user makes some modifications manually.
2. Done. Let me know it its okay or should do it differently.